Argo CD with CodeCommit Contribution

ybezsonov opened the pull request:
#### What this PR does / why we need it:

#### Which issue(s) this PR fixes:

Fixes #376 
Fixes #392 
Fixes #367

#### Quality checks

- [x] My content adheres to the style guidelines
- [x] I ran `make test` or `make e2e-test` and it was successful

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

niallthomson commented on website/docs/automation/gitops/argocd-codecommit/cluster_bootstrap.md:
````diff
@@ -0,0 +1,38 @@
+Let's apply the manifest:
+
+```bash
+$ envsubst < /workspace/modules/automation/gitops/argocd-codecommit/argocd-application.yaml | kubectl apply -f -
````
Lets change this to use kustomize to substitute in the environment variables to make it consistent with all our other labs. I would suggest moving the Argo application in to its own directory (like `environment/workspace/modules/automation/gitops/argocd-codecommit/application`) then apply it with `kubectl apply -k`.

You can see an example of us doing this with a security group policy here:

https://github.com/aws-samples/eks-workshop-v2/tree/main/environment/workspace/modules/networking/securitygroups-for-pods/sg

Please also add the Argo Application CRD to the end of this list so it gets cleaned up by our reset-environment script https://github.com/aws-samples/eks-workshop-v2/blob/main/environment/bin/reset-environment#L43

ybezsonov replied:
Created separate folder for Argo CD kustomization and changed to solution with "replacements" which should be future proof.

ybezsonov replied:
Addedd Argo CD CRD to reset-environment

niallthomson commented on terraform/modules/cluster/addons.tf:
```diff
@@ -236,6 +237,39 @@ module "eks_blueprints_kubernetes_addons" {
     additional_irsa_policies = ["arn:aws:iam::aws:policy/AdministratorAccess"]
   }
 
+  argocd_helm_config = {
```
Can we make sure the Argo components run on the system node group? See this example where we do it for the load balancer controller https://github.com/aws-samples/eks-workshop-v2/blob/main/terraform/modules/cluster/addons.tf#L102

ybezsonov replied:
Used approach from PR 354

niallthomson commented on website/docs/automation/gitops/argocd-codecommit/cleanup.md:
````diff
@@ -0,0 +1,13 @@
+```bash
+$ kubectl -n argocd delete secret codecommit-repo
+secret "codecommit-repo" deleted
+$ kubectl -n argocd delete application apps
````
You'll can remove this once you update the reset-environment script as outlined somewhere else in this review.

ybezsonov replied:
I updated reset-environment, but those resources are created in argocd namespace, so I updated and use explicit clean up here.

niallthomson commented on website/docs/automation/gitops/argocd-codecommit/deploy_application.md:
```diff
@@ -0,0 +1,112 @@
+
+If you need to reset Argo CD `admin` password, the sequence of commands below can be used:
+
+kubectl -n argocd patch secret argocd-secret -p '{"data": {"admin.password": null, "admin.passwordMtime": null}}'
```
You should be able to use a ```bash``` section to wrap these commands so they're a bit easier to see.

ybezsonov replied:
Used `bash` section

niallthomson reviewed (changes requested):
Overall a great foundation for an ArgoCD lab. Appreciate the work put in here and the effort made to align with the other labs. I've made some comments around some of the manifests and Terraform.

ybezsonov commented on terraform/modules/cluster/codecommit.tf:
```diff
@@ -3,6 +3,11 @@ resource "aws_codecommit_repository" "gitops" {
   description     = "CodeCommit repository for GitOps"
 }
 
+resource "aws_codecommit_repository" "gitops-argocd" {
```
fixed

csantanapr commented:
@ybezsonov is the PR ready for another review, and addresses @niallthomson comments?

ybezsonov commented:
> @ybezsonov is the PR ready for another review, and addresses @niallthomson comments?

@csantanapr Yes, I addressed comments and PR passed local make test module="automation/gitops"

csantanapr commented:
@niallthomson when you have a chance could you merge this.

ybezsonov commented:
Hi @niallthomson 
Could you please add e2e-tests label to this PR when it comes to merge?
It passes automated tests in automation/gitops, but it has changes to reset-environment and Dockerfile, so I would like to be sure that it will not  influence other modules.

candonov commented on website/docs/automation/gitops/argocd/update_application.md:
````diff
@@ -0,0 +1,60 @@
+
+To verify, run the following commands:
+
+```bash
+$ kubectl get deployment -n ui ui
+NAME   READY   UP-TO-DATE   AVAILABLE   AGE
+ui     3/3     3            3           3m33s
+$ kubectl get pod -n ui
+NAME                  READY   STATUS    RESTARTS   AGE
+ui-6d5bb7b95-hzmgp   1/1     Running   0          61s
+ui-6d5bb7b95-j28ww   1/1     Running   0          61s
+ui-6d5bb7b95-rjfxd   1/1     Running   0          3m34s
+```
````
Split in two, each command should go in its own code block.

ybezsonov replied:
Done.

candonov reviewed (commented):
The Automation section start with GitOps. ArgoCD should be set to automatically sync the changes when pushed to git. @ybezsonov can you change the application to automatically sync?

ybezsonov commented on website/docs/automation/gitops/argocd/update_application.md:
```diff
@@ -0,0 +1,60 @@
+
+You can review planned changes in the file `/gitops/apps/deployment-patch.yaml`
+
+To apply the patch edit the file `/gitops/apps/kustomization.yaml` like in the example below:
```
Done.

ybezsonov commented on website/docs/automation/gitops/argocd/update_application.md:
````diff
@@ -0,0 +1,60 @@
+automation/gitops/argocd/update-application/kustomization.yaml
+```
+
+You can execute commands to add necessary changes to the file `/gitops/apps/kustomization.yaml`:
````
Done.

ybezsonov commented:
> 

I improved Automation, added AutoSync to the application, described and provide example how to change default Argo CD Sync interval from 3 minutes to short one. Now a learner can get quick feedback on the changes.

candonov commented on environment/workspace/modules/automation/gitops/argocd/update-application/kustomization.yaml:
```diff
@@ -0,0 +1,8 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+bases:
+- ../../../../../manifests/ui
```
Should this be `- ui`?

ybezsonov replied:
This path was required to make kustomizing block show diffs. I changed approach and now it shows the same correct content like in the file which is used for deployment.

candonov commented on website/docs/automation/gitops/argocd/update_application.md:
````diff
@@ -0,0 +1,63 @@
+To apply the patch edit the file `apps/kustomization.yaml` like in the example below:
+
+```file
+automation/gitops/argocd/update-application/kustomization.yaml
````
```
$ cat apps/kustomization.yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
- ui
patches:
- deployment-patch.yaml
```
This relates to the `- ui` line being different in this diff.

ybezsonov replied:
I changed approach and now it shows the same correct content like in the file which is used for deployment.

candonov commented on website/docs/automation/gitops/argocd/deploy_application.md:
````diff
@@ -0,0 +1,103 @@
+
+You should now have all the resources related to the UI services deployed. To verify, run the following commands:
+
+```bash
+$ kubectl get deployment -n ui ui
+NAME   READY   UP-TO-DATE   AVAILABLE   AGE
+ui     1/1     1            1           61s
+$ kubectl get pod -n ui
+NAME                  READY   STATUS    RESTARTS   AGE
+ui-6d5bb7b95-rjfxd   1/1     Running   0          62s
+```
````
Can you split this one two please?

ybezsonov replied:
Done.

candonov reviewed (commented):
Thank you for refactoring the sync. It is great. 
I left couple minor comments and it is ready to merge.

candonov reviewed (dismissed):
/lgtm

candonov commented:
@niallthomson I think you have to ✅ too. Also, the github actions e2e tests fail due to credentials. Do you know why the credentials are not working?

csantanapr commented:
@niallthomson this one is ready for merging

csantanapr commented:
@niallthomson would be good to know what's left on the PR to get it merge, please let @ybezsonov know

niallthomson commented:
Thanks @ybezsonov for the content contribution!
